From fcd6a52b605a4e4b45dabef0a011ee751d93f322 Mon Sep 17 00:00:00 2001
From: Henrik Grimler <grimler@termux.dev>
Date: Sun, 31 Mar 2024 20:20:05 +0200
Subject: [PATCH 35/41] hotspot: os_linux: do not use dlinfo and
 RTLD_DI_LINKMAP on android

As they are not available.
---
 src/hotspot/os/linux/os_linux.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/hotspot/os/linux/os_linux.cpp b/src/hotspot/os/linux/os_linux.cpp
index 115d5345218..bf5330e4331 100644
--- a/src/hotspot/os/linux/os_linux.cpp
+++ b/src/hotspot/os/linux/os_linux.cpp
@@ -1948,10 +1948,12 @@ const char* os::Linux::dll_path(void* lib) {
   const char* l_path = NULL;
   assert(lib != NULL, "dll_path parameter must not be NULL");
 
+#ifndef __ANDROID__
   int res_dli = ::dlinfo(lib, RTLD_DI_LINKMAP, &lmap);
   if (res_dli == 0) {
     l_path = lmap->l_name;
   }
+#endif
   return l_path;
 }
 
-- 
2.50.1

